pkg/controller/render/render_controller.go

```diff
@@ -603,6 +619,33 @@ func generateRenderedMachineConfig(pool *mcfgv1.MachineConfigPool, configs []*mc
 	return merged, nil
 }
 
+// This function takes into account the current MachineConfig that is present
+// on the MachineConfigPool. It first attempts to verify that the newly
+// generated rendered MachineConfig is reconcilable against the current
+// MachineConfig. If it is not reconcilable, the component MachineConfigs that
+// were used to generate the rendered MachineConfig will be evaluated
+// one-by-one to identify which MachineConfig is not reconcilable. The name of
+// the unreconcilable MachineConfig is included in the returned error.
+func generateAndValidateRenderedMachineConfig(currentMC *mcfgv1.MachineConfig, pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.MachineConfig, cconfig *mcfgv1.ControllerConfig) (*mcfgv1.MachineConfig, error) {
+	source := getMachineConfigRefs(configs)
+	klog.V(2).Infof("Considering %d configs %s for MachineConfig generation", len(source), source)
```

Comment:
nit: in practice this seems to be happening a lot: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_machine-config-operator/4158/pull-ci-openshift-machine-config-operator-master-e2e-gcp-op/1757856090667618304/artifacts/e2e-gcp-op/gather-extra/artifacts/pods/openshift-machine-config-operator_machine-config-controller-6b6c5c865b-f7cz2_machine-config-controller.log , would you consider dropping this to V(4)?

(I guess a lot is relative, since we do have a lot of logs in there, maybe not too bad to have this as well)

Reply:
Yeah, I'll change that to `V(4)`. I purposely made it more verbose for testing purposes. :smile: